grails-app/services/org/pih/warehouse/fulfillment/FulfillmentService.groovy

```diff
@@ -378,4 +379,71 @@ class FulfillmentService {
 
         return debitTransaction
     }
+
+    String bindOrInferLotNumber(ImportPackingListItem obj, Map source) {
+        String lotNumber = source['lotNumber']
+        String productCode = source['product']
+        String binLocationName = source['binLocation']
+
+        // If lot number is provided then just return this value and bind it
+        // Otherwise try to infer the lot number
+        if (lotNumber) {
+            return lotNumber
+        }
+
+        // If bin location is not provided then first check if inventory with default lot number is available
+        if (!binLocationName) {
+            AvailableItem itemsWithDefaultLot = productAvailabilityService
+                    .getAvailableItemWithDefaultLots(obj.origin, productCode)
+            // only infer if there is one possible value
+            if (itemsWithDefaultLot) {
+                return itemsWithDefaultLot?.inventoryItem?.lotNumber
+            }
+        }
+
+        // otherwise try to infer lotNumber based on provided binLocation
+        AvailableItem availableItem = productAvailabilityService
+                .inferAvailableItemByBinLocation(obj.origin, productCode, binLocationName)
+
+        return availableItem?.inventoryItem?.lotNumber
+    }
+
+    Location bindOrInferBinLocation(ImportPackingListItem obj, Map source) {
+        String lotNumber = source['lotNumber']
+        String productCode = source['product']
+        String binLocationName = source['binLocation']
+
+        ProductAvailabilityService productAvailabilityService = Holders.grailsApplication.mainContext.getBean(ProductAvailabilityService)
```

Comment:
This service is already injected in the `FulfillmentService`, you don't have to inject it again here.